dot/digest/digest_test.go

```diff
@@ -420,9 +419,34 @@ func TestHandler_HandleNextEpochData(t *testing.T) {
 
 	header := createHeaderWithPreDigest(t, 10)
 
+	finalisedCh := make(chan *types.FinalisationInfo)
+
+	handler := newTestHandler(t)
+	handler.finalised = finalisedCh
+
+	ctx, cancel := context.WithCancel(context.Background())
+
+	doneCh := make(chan struct{})
+
+	go func() {
+		defer close(doneCh)
+		handler.handleBlockFinalisation(ctx)
+	}()
```

Comment:
This does not work, `handleBlockFinalisation` contains:
````
for {
		select {
		case info := <-h.finalised:
                 case <-ctx.Done():
}
```
And if I send the data through `finalisedCh` before synchronously calling `handler.handleBlockFinalisation(ctx)`, then it will block since the `h.finalised` keeps open and the `<-ctx.Done()` is not trigged yet

Reply:
An alternative is: creating a `context.WithTimeout` and passing it to the `handler.handleBlockFinalisation(ctx)` and then wait for the `<-ctx.Done()` be trigged